File: packages/java/anip-core/src/main/java/dev/anip/core/CheckpointDetailResponse.java
package dev.anip.core;

import java.util.Map;

/**
 * Response from the checkpoint detail endpoint.
 */
public class CheckpointDetailResponse {

    private final Map<String, Object> checkpoint;
    private final Map<String, Object> inclusionProof;
    private final String proofUnavailable;

    public CheckpointDetailResponse(Map<String, Object> checkpoint,
                                     Map<String, Object> inclusionProof,
                                     String proofUnavailable) {
        this.checkpoint = checkpoint;
        this.inclusionProof = inclusionProof;
        this.proofUnavailable = proofUnavailable;
    }

    public Map<String, Object> getCheckpoint() {
        return checkpoint;
    }

    public Map<String, Object> getInclusionProof() {
        return inclusionProof;
    }

    public String getProofUnavailable() {
        return proofUnavailable;
    }
}

File: packages/csharp/src/Anip.Core/CheckpointDetailResponse.cs
using System.Text.Json.Serialization;

namespace Anip.Core;

public class CheckpointDetailResponse
{
    [JsonPropertyName("checkpoint")]
    public Dictionary<string, object>? Checkpoint { get; set; }

    [JsonPropertyName("inclusion_proof")]
    public Dictionary<string, object>? InclusionProof { get; set; }

    [JsonPropertyName("consistency_proof")]
    public Dictionary<string, object>? ConsistencyProof { get; set; }

    [JsonPropertyName("proof_unavailable")]
    public string? ProofUnavailable { get; set; }

    [JsonPropertyName("expires_hint")]
    public string? ExpiresHint { get; set; }
}
